### PR TITLE
Refactoring before making Kotlin DSL models IP compatible

### PR DESCRIPTION
**Pull request by alllex**
This is only clean up and polishing, no semantic changes

Preparation for:
- https://github.com/gradle/gradle/pull/27176

### CODE REVIEW

**Comment by 6hundreds** [comment]
Looks good 👍

**Review by 6hundreds — APPROVED** [review]

**Comment by alllex** [comment]
@bot-gradle merge

File: platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinDslScriptsModelBuilder.kt [hunk 1]
```diff
@@ -109,8 +109,11 @@ data class StandardEditorPosition(
 internal
 object KotlinDslScriptsModelBuilder : ToolingModelBuilder {
 
+    private
+    val MODEL_NAME = KotlinDslScriptsModel::class.qualifiedName!!
```
Comment (bamboo) [inline platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinDslScriptsModelBuilder.kt:113]: ❌ Let's avoid Kotlin reflection just to answer `canBuild`

File: platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinDslScriptsModelBuilder.kt [hunk 2]
```diff
@@ -188,44 +191,56 @@ fun dehydrateScriptModels(
...
-    val extension = ".gradle.kts"
-
     // Init Scripts
     project
         .gradle
         .startParameter
         .allInitScripts
-        .filter(File::isFile)
-        .filter { it.name.endsWith(extension) }
+        .filter { it.isKotlinDslFile }
```
Comment (bamboo) [inline platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinDslScriptsModelBuilder.kt:231]: 👍

File: subprojects/core/src/main/java/org/gradle/tooling/provider/model/internal/IntermediateToolingModelProvider.java [hunk 3]
```diff
@@ -37,15 +38,15 @@ public interface IntermediateToolingModelProvider {
     /**
      * Fetches models of a given type for the given projects passing a parameter to the underlying builder.
      */
-    <T> List<T> getModels(Project requester, List<Project> targets, String modelName, Class<T> modelType, Object modelBuilderParameter);
+    <T> List<T> getModels(Project requester, List<Project> targets, String modelName, Class<T> modelType, @Nullable Object parameter);
```
Comment (bamboo) [inline subprojects/core/src/main/java/org/gradle/tooling/provider/model/internal/IntermediateToolingModelProvider.java:41]: 👍

**Review by bamboo — CHANGES_REQUESTED** [review]

**Comment by alllex** [comment]
@bot-gradle help

**Comment by alllex** [comment]
@bot-gradle cancel

File: platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinBuildScriptModelBuilder.kt [hunk 4]
```diff
@@ -96,8 +96,11 @@ data class StandardKotlinBuildScriptModel(
 internal
 object KotlinBuildScriptModelBuilder : ToolingModelBuilder {
 
+    private
+    val MODEL_NAME = KotlinBuildScriptModel::class.qualifiedName!!
+
     override fun canBuild(modelName: String): Boolean =
-        modelName == "org.gradle.kotlin.dsl.tooling.models.KotlinBuildScriptModel"
+        modelName == MODEL_NAME
```
[the remark this answers is not in the document]
Comment (alllex) [reply platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinBuildScriptModelBuilder.kt]: Fixed

File: platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinBuildScriptModelBuilder.kt [hunk 5]
```diff
@@ -251,7 +254,9 @@ fun projectScriptModelBuilder(
         val stage1BlocksAccessorClassPathGenerator = project.serviceOf<Stage1BlocksAccessorClassPathGenerator>()
         val projectAccessorClassPathGenerator = project.serviceOf<ProjectAccessorsClassPathGenerator>()
         val dependenciesAccessors = project.serviceOf<DependenciesAccessors>()
-        projectAccessorClassPathGenerator.projectAccessorsClassPath(project, classPath) + stage1BlocksAccessorClassPathGenerator.stage1BlocksAccessorClassPath(project) + AccessorsClassPath(dependenciesAccessors.classes, dependenciesAccessors.sources)
+        projectAccessorClassPathGenerator.projectAccessorsClassPath(project, classPath) +
```
[the remark this answers is not in the document]
Comment (alllex) [reply platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinBuildScriptModelBuilder.kt]: Good idea. Fixed

**Review by eskatos — APPROVED** [review]

File: platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinBuildScriptModelBuilder.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (alllex) [reply platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinBuildScriptModelBuilder.kt]: Kotlin does not seem to allow that, giving a compilation error. It thinks that the previous statement has finished, and can't resolve an unary plus operator

[the remark this answers is not in the document]
Comment (alllex) [reply platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinBuildScriptModelBuilder.kt]: I moved the expression into a separate function, so it's easier to read

**Comment by alllex** [comment]
@bot-gradle merge

File: platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinBuildScriptModelBuilder.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (bamboo) [reply platforms/core-configuration/kotlin-dsl-tooling-builders/src/main/kotlin/org/gradle/kotlin/dsl/tooling/builders/KotlinBuildScriptModelBuilder.kt]: Right, Kotlin requires parenthesizing the whole expression

**Review by bamboo — APPROVED** [review]

**Pull request merged**